sw/device/lib/dif/dif_spi_host_unittest.cc

```diff
@@ -0,0 +1,517 @@
+                     {SPI_HOST_CONFIGOPTS_CSNTRAIL_0_OFFSET, 0},
+                     {SPI_HOST_CONFIGOPTS_CSNLEAD_0_OFFSET, 0},
+                     {SPI_HOST_CONFIGOPTS_FULLCYC_0_BIT, false},
+                     {SPI_HOST_CONFIGOPTS_CPHA_0_BIT, false},
+                     {SPI_HOST_CONFIGOPTS_CPOL_0_BIT, false},
+                 });
+  EXPECT_WRITE32(SPI_HOST_CONTROL_REG_OFFSET,
+                 {
+                     {SPI_HOST_CONTROL_SPIEN_BIT, true},
+                 });
+
+  EXPECT_EQ(dif_spi_host_configure(&spi_host_, config_), kDifOk);
+}
+
+// Checks that the chip select options get written to the appropriate fields in
+// the config register.
+TEST_F(ConfigTest, ChipSelectOptions) {
+  config_.chip_select.idle = 1;
+  config_.chip_select.trail = 2;
+  config_.chip_select.lead = 3;
+
+  ExpectDeviceReset();
+  EXPECT_WRITE32(SPI_HOST_CONFIGOPTS_REG_OFFSET,
+                 {
+                     {SPI_HOST_CONFIGOPTS_CLKDIV_0_OFFSET, 0},
+                     {SPI_HOST_CONFIGOPTS_CSNIDLE_0_OFFSET, 1},
+                     {SPI_HOST_CONFIGOPTS_CSNTRAIL_0_OFFSET, 2},
+                     {SPI_HOST_CONFIGOPTS_CSNLEAD_0_OFFSET, 3},
+                     {SPI_HOST_CONFIGOPTS_FULLCYC_0_BIT, false},
+                     {SPI_HOST_CONFIGOPTS_CPHA_0_BIT, false},
+                     {SPI_HOST_CONFIGOPTS_CPOL_0_BIT, false},
+                 });
+  EXPECT_WRITE32(SPI_HOST_CONTROL_REG_OFFSET,
+                 {
+                     {SPI_HOST_CONTROL_SPIEN_BIT, true},
+                 });
+
+  EXPECT_EQ(dif_spi_host_configure(&spi_host_, config_), kDifOk);
+}
+
+// Checks that the SPI cycle, polarity and phase options get written to the
+// appropriate fields in the config register.
+TEST_F(ConfigTest, SpiOptions) {
+  config_.full_cycle = true;
+  config_.cpol = true;
+  config_.cpha = true;
+
+  ExpectDeviceReset();
+  EXPECT_WRITE32(SPI_HOST_CONFIGOPTS_REG_OFFSET,
+                 {
+                     {SPI_HOST_CONFIGOPTS_CLKDIV_0_OFFSET, 0},
+                     {SPI_HOST_CONFIGOPTS_CSNIDLE_0_OFFSET, 0},
+                     {SPI_HOST_CONFIGOPTS_CSNTRAIL_0_OFFSET, 0},
+                     {SPI_HOST_CONFIGOPTS_CSNLEAD_0_OFFSET, 0},
+                     {SPI_HOST_CONFIGOPTS_FULLCYC_0_BIT, true},
+                     {SPI_HOST_CONFIGOPTS_CPHA_0_BIT, true},
+                     {SPI_HOST_CONFIGOPTS_CPOL_0_BIT, true},
+                 });
+  EXPECT_WRITE32(SPI_HOST_CONTROL_REG_OFFSET,
+                 {
+                     {SPI_HOST_CONTROL_SPIEN_BIT, true},
+                 });
+
+  EXPECT_EQ(dif_spi_host_configure(&spi_host_, config_), kDifOk);
+}
+
+class TransactionTest : public SpiHostTest {
+ protected:
+  MockFifo fifo_;
+};
+
+// Checks that an opcode segment is sent correctly.
+TEST_F(TransactionTest, IssueOpcode) {
+  dif_spi_host_segment segments[] = {
+      {
+          .type = kDifSpiHostSegmentTypeOpcode,
+          .opcode = 0x5a,
+      },
```

Comment:
Nit: for single-element arrays, skipping the trailing comma will cause the compiler to group the `{{` together.

Reply:
Done... at least for some of these.  `clang-format` has some strange ideas about the layout for some of these.